### PR TITLE
wiggle: GuestPtr<[u8]> => GuestPtr<str> conversions 

### LINKED ISSUE #1888: wiggle: GuestPtr<[u8]> <=> GuestPtr<str> conversions
This issue is specific to the Wiggle crate. It is a good beginner issue.

#### Feature

We don't have a way to convert between `GuestPtr<[u8]>` and `GuestPtr<str>`. I just came across a case where it would be useful - I have a `GuestPtr<str>` and I need to pass the underlying data as `&[u8]` to a function that will perform its own utf8 validation - and the opposite is probably useful as well.

#### Implementation

Add methods to `impl GuestPtr<[u8]>` and impl `GuestPtr<str>` that return the other type.

### PR DESCRIPTION
**Pull request by siyopao**
As requested in #1888, adds a method to convert a `GuestPtr<[u8]>`  to  a `GuestPtr<str>` without first validating that the `[u8]` consists of utf-8 codepoints. The conversion in the other direction already exists as `as_bytes`.

Although unrelated to this PR, while I was in there I found a couple of small typos in the docs so I fixed them.

Resolves #1888.

### CODE REVIEW

**Comment by pchickey** [comment]
Thanks! Can you also add the corresponding GuestPtr<str> -> GuestPtr<[u8]> conversion?

**Comment by siyopao** [comment]
Hello @pchickey. Thank you! Okay, I added the `as_byte_ptr(&self) -> GuestPtr<'a, [u8]>` method. It is exactly the same as the `as_bytes(&self) -> GuestPtr<'a, [u8]>` method. Is that the intention?

File: crates/wiggle/src/lib.rs [hunk 1]
```diff
@@ -499,6 +499,12 @@ impl<'a> GuestPtr<'a, str> {
         GuestPtr::new(self.mem, self.pointer)
     }
 
+    /// Returns a raw pointer for the underlying slice of bytes that this
```
Comment (pchickey) [inline crates/wiggle/src/lib.rs:502]:
```suggestion
    /// Returns a pointer for the underlying slice of bytes that this
```

File: crates/wiggle/src/lib.rs [hunk 2]
```diff
@@ -533,6 +539,14 @@ impl<'a> GuestPtr<'a, str> {
     }
 }
 
+impl<'a> GuestPtr<'a, [u8]> {
+    /// Returns a raw pointer to the string represented by a `[u8]` without
```
Comment (pchickey) [inline crates/wiggle/src/lib.rs:543]:
```suggestion
    /// Returns a pointer to the string represented by a `[u8]` without
```

**Comment by siyopao** [comment]
Thank you for the explanation and your help @pchickey. It was an interesting first look!

**Pull request merged**